debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/memory/MemoryLogMinerEventProcessor.java

```diff
@@ -273,33 +285,38 @@ protected Scn calculateNewStartScn(Scn endScn, Scn maxCommittedScn) throws Inter
 
     @Override
     protected Scn getTransactionCacheMinimumScn() {
-        return transactionCache.values().stream()
-                .map(MemoryTransaction::getStartScn)
-                .min(Scn::compareTo)
-                .orElse(Scn.NULL);
+        AtomicReference<Scn> result = new AtomicReference<>();
```

Comment:
Mostly an FYI, I don't mind the use of atomics, my only concern is their impact on performance, especially if used within the context of a processing loop.  If it's once per iteration, then I would say those are generally okay.